tasks/eva_3033/detect_project_status.py

```diff
@@ -0,0 +1,228 @@
+        """
+        accessioning_reports = self.get_accession_reports_for_study()
+        accessioned_filenames = [self.get_accession_file(f) for f in filenames]
+        if not accessioning_reports:
+            return 0, []
+        if len(accessioning_reports) == 1:
+            accessioning_report = accessioning_reports[0]
+        elif len([r for r in accessioning_reports if r in accessioned_filenames]) == 1:
+            accessioning_report = [r for r in accessioning_reports if os.path.basename(r) in accessioned_filenames][0]
+        elif len([r for r in accessioning_reports if analysis in r]) == 1:
+            accessioning_report = [r for r in accessioning_reports if analysis in r][0]
+        elif accessioning_reports:
+            logger.warning(f'Assume all accessioning report are from project {self.project}:{analysis} and only use the first one.')
+            accessioning_report = accessioning_reports[0]
+        else:
+            logger.error(f'Cannot assign accessioning report to project {self.project} analysis {analysis} for files {accessioning_reports}')
+            accessioning_report = None
+        if not accessioning_report:
+            return 0, []
+        return self.get_accessioning_info_from_file(accessioning_report)
+
+    def get_accession_file(self, filename):
+        basefile = ''
+        if filename.endswith('.vcf.gz'):
+            basefile = filename[:-7]
+        elif filename.endswith('.vcf'):
+            basefile = filename[:-4]
+        return basefile + '.accessioned.vcf.gz '
+
+    def get_accessioning_info_from_file(self, path):
+        """
+        Read the accessioning report to retrieve the first 1000 Submitted variant accessions
+        """
+        no_of_ss_ids_in_file = 0
+        first_1000_ids = []
+        with gzip.open(path, 'rt') as f:
+            for line in f:
+                if line.startswith('#'):
+                    continue
+                elif line.split("\t")[2].startswith("ss"):
+                    no_of_ss_ids_in_file = no_of_ss_ids_in_file + 1
+                    if no_of_ss_ids_in_file <= 1000:
+                        first_1000_ids.append(int(line.split("\t")[2][2:]))
+                    else:
+                        break
+        return first_1000_ids
+
+    def get_accession_reports_for_study(self):
+        """
+        Given a study, find the accessioning report path for that study on both noah and codon
+        look for a file ending with accessioned.vcf.gz (assuming only one file with the given pattern will be present)
+        if more than one files are present, take the first one
+        """
+
+        noah_files = glob.glob(os.path.join(self.noah_project_dir, self.project, '60_eva_public', '*accessioned.vcf.gz'))
+        codon_files = glob.glob(os.path.join(self.codon_project_dir, self.project, '60_eva_public', '*accessioned.vcf.gz'))
+        accessioning_reports = noah_files + codon_files
```

Comment:
I am wondering if we should deduplicate here just in case there are any duplicates...